### PR TITLE
Discussion: add `update_time` column to source state table

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

To improve observability and facilitate the measurement of source freshness
related: https://github.com/risingwavelabs/risingwave-docs/issues/1513

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by lmatz — APPROVED** [review]
LGTM, thanks, cc: @cyliu0 

so for each input row, there will be a unique timestamp generated for it, right?

**Comment by StrikeW** [comment]
> LGTM, thanks, cc: @cyliu0
> 
> so for each input row, there will be a unique timestamp generated for it, right?

Precisely, the row can be unchanged as before since we employ the heartbeat event of debezium, so if upstream database hasn't been update for a long time, the offset in the heartbeat will remain the same.

**Comment by lmatz** [comment]
Just leave a comment for visibility

I think it is okay to use `update_time` as the metric that we track in our internal test pipelines.

i.e. during PG -> CDC -> RW, after all of the data has been updated to PG and `all` of its CDC has been synced to RW, we get the `commit_ts` from PG and `update_time` from RW, and we compare the difference between them to show whether RW was lagging behind PG.

But one problem is that it seems the user cannot use this trick to monitor a pipeline where the PG still gets updated and CDCed as there is no notion of `all`.

Moreover, if `offset` is a more intuitive concept for users to monitor if RW is lagging, I feel we'd better use the same metric, e.g. `offset`, in our internal testing pipeline.


`commit_ts` synced to RW and minus by `update_time` seems a more intuitive approach.

File: src/frontend/src/optimizer/plan_node/generic/source.rs [hunk 1]
```diff
@@ -117,9 +117,16 @@ impl Source {
             sub_fields: vec![],
             type_name: "".to_string(),
         };
+        let update_time = Field {
```
Comment (hzxa21) [inline src/frontend/src/optimizer/plan_node/generic/source.rs:120]: Will this affect the schema of the already created source/table? Is this change backward-compatible?

Comment (hzxa21) [reply src/frontend/src/optimizer/plan_node/generic/source.rs:120]: If not, I suggest we add the update time field in the json.

Comment (StrikeW) [reply src/frontend/src/optimizer/plan_node/generic/source.rs:120]: This pr won't merge, yesterday we found that debezium can capture the event time on upstream, so we decide to generate a (RW process time - event time) metric to promethues as an indication of lagging. FYI
https://github.com/risingwavelabs/risingwave/issues/13440

**Pull request closed without merging**